pylutron_caseta/leap.py

```diff
@@ -3,21 +3,21 @@
 import asyncio
 import json
 import logging
+import re
 
 _LOG = logging.getLogger(__name__)
 _DEFAULT_LIMIT = 2 ** 16
 
 
-@asyncio.coroutine
-def open_connection(host=None, port=None, *,
-                    loop=None, limit=_DEFAULT_LIMIT, **kwds):
+async def open_connection(host=None, port=None, *,
+                          limit=_DEFAULT_LIMIT, **kwds):
     """Open a stream and wrap it with LEAP."""
-    connection = yield from asyncio.open_connection(host, port, loop=loop,
-                                                    limit=limit, **kwds)
+    connection = await asyncio.open_connection(host, port,
+                                               limit=limit, **kwds)
```

Comment:
It might cause problems on early (buggy) versions of Python 3.5, but I don't know exactly what changed.  I don't expect it to cause problems, since I would expect `asyncio.open_connection` to run in the loop that called it.  Certainly there are asyncio unit tests that exercise this (most common) case.

The only possible problem with this change that the connection is opened in the wrong loop (or no loop).  However, it's clearly executed in a loop (since it's awaited), and the intention in this code is to run it in the same event loop that the parent method was executed in.

I don't think there's really any loop ambiguity in this library, so I don't think there's much risk to this change.